plugins/module_utils/k8s/runner.py

```diff
@@ -0,0 +1,122 @@
+# Copyright: (c) 2021, Red Hat | Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from typing import Dict
+
+from ansible_collections.kubernetes.core.plugins.module_utils.k8s.client import (
+    get_api_client,
+)
+from ansible_collections.kubernetes.core.plugins.module_utils.k8s.resource import (
+    create_definitions,
+)
+from ansible_collections.kubernetes.core.plugins.module_utils.k8s.service import (
+    K8sService,
+)
+from ansible_collections.kubernetes.core.plugins.module_utils.k8s.exceptions import (
+    CoreException,
+)
+
+from ansible_collections.kubernetes.core.plugins.module_utils.selector import (
+    LabelSelectorFilter,
+)
+
+try:
+    from kubernetes.dynamic.exceptions import KubernetesValidateMissing
+except ImportError:
+    # Handled in module setup
+    pass
+
+
+def validate(client, module, resource):
+    def _prepend_resource_info(resource, msg):
+        return "%s %s: %s" % (resource["kind"], resource["metadata"]["name"], msg)
+
+    try:
+        warnings, errors = client.validate(
+            resource,
+            module.params["validate"].get("version"),
+            module.params["validate"].get("strict"),
+        )
+    except KubernetesValidateMissing:
+        module.fail_json(
+            msg="kubernetes-validate python library is required to validate resources"
+        )
+
+    if errors and module.params["validate"]["fail_on_error"]:
+        module.fail_json(
+            msg="\n".join([_prepend_resource_info(resource, error) for error in errors])
+        )
+    else:
+        return [_prepend_resource_info(resource, msg) for msg in warnings + errors]
```

Comment:
```suggestion
    if errors and module.params["validate"]["fail_on_error"]:
        module.fail_json(
            msg="\n".join([_prepend_resource_info(resource, error) for error in errors])
        )
    return [_prepend_resource_info(resource, msg) for msg in warnings + errors]
```

Reply:
@Akasurde Thank you. Done! This requires https://github.com/ansible-collections/kubernetes.core/pull/307 to be merged first.